ga4gh/backend.py

```diff
@@ -215,6 +217,68 @@ def variantSetsGenerator(self, request):
         return self._topLevelObjectGenerator(
             request, self._variantSetIdMap, self._variantSetIds)
 
+    def readsGenerator(self, request):
+        # Local utility functions to save some typing
+        def getPosition(readAlignment):
+            return readAlignment.alignment.position.position
+
+        def getEndPosition(readAlignment):
+            return getPosition(readAlignment) + \
+                len(readAlignment.alignedSequence)
+
+        if len(request.readGroupIds) != 1:
+            raise exceptions.NotImplementedException(
+                "Read search over multiple readGroups not supported")
+        readGroupId = request.readGroupIds[0]
+        try:
+            readGroup = self._readGroupIdMap[request.readGroupIds[0]]
+        except KeyError:
+            raise exceptions.ReadGroupNotFoundException(readGroupId)
+        startPosition = request.start
+        equalPositionsToSkip = 0
+        if request.pageToken is not None:
+            startPosition, equalPositionsToSkip = self.parsePageToken(
+                request.pageToken, 2)
+        iterator = readGroup.getReadAlignments(
+            request.referenceName, request.referenceId, startPosition,
+            request.end)
+        readAlignment = next(iterator, None)
+        if request.pageToken is not None:
+            # First, skip any reads with position < startPosition
+            # or endPosition < startPosition
+            while (getPosition(readAlignment) < startPosition or
+                   getEndPosition(readAlignment) < startPosition):
```

Comment:
Since `startPosition` here may already be adjusted to something else than `request.start`, this line will need to read:
```
getEndPosition(readAlignment) < request.start):
```